client/src/partials/cash/modals/transfer.modal.js

```diff
@@ -0,0 +1,113 @@
+angular.module('bhima.controllers')
+.controller('CashTransferModalController', CashTransferModalController);
+
+CashTransferModalController.$inject = [
+  '$uibModalInstance', '$q', 'SessionService', 'CurrencyService', 'VoucherService', 'CashboxService', 'AccountService', 'cashBox', 'uuid'
+];
+
+/**
+ * @module cash/modals/CashTransferModalController
+ *
+ * @description This controller is responsible transfering money between auxillary cash and a virement account
+*/
+
+function CashTransferModalController(ModalInstance, $q, sessionService, currencyService, voucherService, cashBoxService, accountService, cashBox, uuid) {
+  var vm = this; 
+
+  /** Attaching service to the scope **/
+  vm.cashBox = cashBox;
+  vm.currencyService = currencyService; 
+  vm.cashBoxService = cashBoxService;
+
+  /** check if the cashbox id is provided**/
+  vm.cashBoxIdMissed = !cashBox.id;
+
+  /**
+  * @function submit
+  * @param {boolean} invalid it is true if the form is valid
+  * @description for submitting a dialog content
+  **/
+  function submit (invalid){
+    if (invalid) { return; }
+    var voucher = processVoucher();
+    var voucher_item = processVoucherItem(voucher.uuid);
+    var record = { voucher : voucher, voucher_item : voucher_item };
+    voucherService.create(record)      
+    .then(function (res){
+      vm.submited = true; //FIX ME, I need a goo idea
+    })
+    .catch(function (err){
+      vm.fail = true; //FIX ME, I need a goo idea
```

Comment:
In my opinion, it is better to show the translated error code to the user.  That is why we have `req.codes` on the server.  This enables us to show the user any server sent error message, including:
 1. "No connection to the server."
 2. "You are missing a value for column 'date'."
 3. "The server is out of memory"
 4. "No response from the database"
 5. "Unknown error, please try again."
 6. "The data you are editing has been modified.  Please refresh the page and try again."

These are complicated, but all require different actions.  We want the user to be informed about what actions they can take to resolve any problems that occur.

 The main problem with the old `messenger` service was that it just displayed a massive red text that didn't mean anything.  The user did not know how to solve the problem and neither did we.